src/tox_current_env/hooks.py

```diff
@@ -1,260 +1,6 @@
-        if sys.platform == "win32":
-            # Avoid requiring admin rights on Windows
-            subprocess.check_call(f'mklink /J "{link}" "{target}"', shell=True)
-        else:
-            os.symlink(target, link)
-        # prevent tox from creating the venv
-        return True
-    if not is_proper_venv(venv):
-        rm_venv(venv)
-    return None  # let tox handle the rest
-
-
-@tox.hookimpl
-def tox_package(session, venv):
-    """Fail early when unsupported"""
-    config = venv.envconfig.config
-    unsupported_raise(config, venv)
-
-
-@tox.hookimpl
-def tox_testenv_install_deps(venv, action):
-    """We don't install anything"""
-    config = venv.envconfig.config
-    unsupported_raise(config, venv)
-    if _plugin_active(config.option):
-        return True
-
-
-def tox_dependencies(config):
-    """Get dependencies of tox itself, 'minversion' and 'requires' config options"""
-    if config.minversion is not None:
-        yield f"tox >= {config.minversion}"
-    # config does not have the "requires" attribute until tox 3.22:
-    yield from getattr(config, "requires", [])
-
-
-@tox.hookimpl
-def tox_runtest(venv, redirect):
-    """If --print-deps-to, prints deps instead of running tests.
-    If --print-extras-to, prints extras instead of running tests.
-    Both options can be used together."""
-    config = venv.envconfig.config
-    unsupported_raise(config, venv)
-    ret = None
-
-    if config.option.print_deps_to:
-        print(
-            *tox_dependencies(config),
-            *venv.get_resolved_dependencies(),
-            sep="\n",
-            file=config.option.print_deps_to,
-        )
-        config.option.print_deps_to.flush()
-        ret = True
-
-    if config.option.print_extras_to:
-        print(
-            *venv.envconfig.extras,
-            sep="\n",
-            file=config.option.print_extras_to,
-        )
-        config.option.print_extras_to.flush()
-        ret = True
-
-    return ret
-
-
-@tox.hookimpl
-def tox_cleanup(session):
-    """Remove the fake virtualenv not to collide with regular tox
-    Collisions can happen anyway (when tox is killed forcefully before this happens)
-    Note that we don't remove real venvs, as recreating them is expensive"""
-    for venv in session.venv_dict.values():
-        if is_current_env_link(venv):
-            rm_venv(venv)
-
-
-@tox.hookimpl
-def tox_runenvreport(venv, action):
-    """Prevent using pip to display installed packages,
-    use importlib.metadata instead, but fallback to default without our flags."""
-    if not _plugin_active(venv.envconfig.config.option):
-        return None
-    return (
-        "{}=={}".format(d.metadata.get("name"), d.version)
-        for d in sorted(
-            importlib_metadata.distributions(), key=lambda d: d.metadata.get("name")
-        )
-    )
+if TOX_VERSION[0] == "4":
+    from tox_current_env.hooks4 import *
+else:
+    from tox_current_env.hooks3 import *
```

Comment:
Ugh, can we not mix the new and old please? Makes the code hard to read and the design awkward trying to have your fingers in two pies. IMHO a better approach would be to just start from scratch for tox 4 and release as version 2 :shrug:

Reply:
From the discussion with my colleagues, we agreed that we'd like to keep only one maintained version with the maintenance burden low.

Reply:
So drop version 3 support?🤣

Reply:
We cannot do that until we update tox to 4 in all supported Fedora releases.